### PR TITLE
Change `normalized` to `SphericalHarmonics` and `SolidHarmonics` across all APIs

### PR DESCRIPTION
**Pull request by frostedoyster**
This is much more intuitive for the user (at the moment, if they ask for the spherical harmonics, they get something else by default), matches the Julia API, and also seems to be accepted in the field (see e3x API for spherical and solid harmonics)

### CODE REVIEW

**Comment by frostedoyster** [comment]
I can't get torch to work, I think the inheritance `torch::CustomClassHolder` -> `SphericalHarmonics` -> `SolidHarmonics` is breaking the `CustomClassHolder` mechanics.
Multiple inheritance of `SolidHarmonics` from `SphericalHarmonics` and  `CustomClassHolder` doesn't compile (understandably perhaps).
@Luthaf please have a look if and when you can

**Comment by frostedoyster** [comment]
This would solve #98

**Comment by Luthaf** [comment]
> I can't get torch to work, I think the inheritance torch::CustomClassHolder -> SphericalHarmonics -> SolidHarmonics is breaking the CustomClassHolder mechanics.
Multiple inheritance of SolidHarmonics from SphericalHarmonics and CustomClassHolder doesn't compile (understandably perhaps).

Yes, you'll need multiple classes at the Torch API level, one for each calculator.

**Comment by frostedoyster** [comment]
Torch should be good. The docs are not ideal for now, but I'm planning to do a thorough docs revision PR after #139 is fixed

File: examples/pytorch/example.py [hunk 1]
```diff
@@ -19,7 +19,10 @@ class SHModule(torch.nn.Module):
     `torch.nn.Module`"""
 
     def __init__(self, l_max, normalized=False):
-        self.spherical_harmonics = sphericart.torch.SphericalHarmonics(l_max, normalized)
+        if normalized:
+            self.spherical_harmonics = sphericart.torch.SphericalHarmonics(l_max)
+        else:
+            self.spherical_harmonics = sphericart.torch.SolidHarmonics(l_max)
```
Comment (Luthaf) [inline examples/pytorch/example.py:25]: Does it make sense to keep both options here now there there is no longer a normalized option? I would only demonstrate one of the calculator in the example and not muddy the water with more option than required

File: sphericart/include/sphericart.h [hunk 2]
```diff
@@ -13,8 +13,10 @@
 
 #include "sphericart.hpp"
 
-using sphericart_calculator_t = sphericart::SphericalHarmonics<double>;
-using sphericart_calculator_f_t = sphericart::SphericalHarmonics<float>;
+using sphericart_spherical_harmonics_calculator_t = sphericart::SphericalHarmonics<double>;
+using sphericart_spherical_harmonics_calculator_f_t = sphericart::SphericalHarmonics<float>;
+using sphericart_solid_harmonics_calculator_t = sphericart::SolidHarmonics<double>;
+using sphericart_solid_harmonics_calculator_f_t = sphericart::SolidHarmonics<float>;
```
Comment (Luthaf) [inline sphericart/include/sphericart.h:19]: Do we really need separate types here? We could reduce the number of functions by either having a single calculator type, which is initialized to be either `SphericalHarmonics` or `SolidHarmonics`, or having two calculator types by a single function that can take both and call the right function on them.

If this is not an issue, we could also keep the two separate, but I feel like we have a lot of functions in the C API for no reason.

File: examples/pytorch/example.py (hunk 1 above)
Comment (frostedoyster) [reply examples/pytorch/example.py:25]: Yes, let's just keep the SphericalHarmonics

File: sphericart/include/sphericart.h (hunk 2 above)
Comment (frostedoyster) [reply sphericart/include/sphericart.h:19]: I agree that we have a lot of functions (and we would have even more if we wanted to have a CUDA C API for e.g. CuPy). However, I feel like this is in line with the other APIs and the original idea behind this PR was to make the APIs as uniform as possible

**Review by Luthaf — APPROVED** [review]

File: sphericart/include/sphericart.h (hunk 2 above)
Comment (frostedoyster) [reply sphericart/include/sphericart.h:19]: Thanks. And yes, let's discuss in person
